packages/headless/src/features/search/search-actions.ts

```diff
@@ -147,14 +157,51 @@ export const executeSearch = createAsyncThunk<
   }
 );
 
+export const fetchMoreResults = createAsyncThunk<
+  ExecuteSearchThunkReturn,
+  void,
+  AsyncThunkSearchOptions<StateNeededByExecuteSearch>
+>(
+  'search/fetchMoreResults',
+  async (
+    _,
+    {getState, dispatch, rejectWithValue, extra: {searchAPIClient}}
+  ) => {
+    const state = getState();
+    addEntryInActionsHistory(state);
```

Comment:
I would not add that as an entry in actions history, because it's not a "new search", but merely a continuation of an old query.

Which makes me realize we might have a bug for any `executeSearch` performed by the pager, as the same logic would apply.

I'll open another JIRA to investigate.

Reply:
Good point